class.jetpack-cli.php

```diff
@@ -291,22 +302,95 @@ public function reset( $args, $assoc_args ) {
+					Jetpack_Options::update_option( 'jumpstart', 'new_connection' );
+				}
 				WP_CLI::success( __( 'jumpstart option reset', 'jetpack' ) );
 				break;
 			case 'modules':
-				$default_modules = Jetpack::get_default_modules();
-				Jetpack::update_active_modules( $default_modules );
+				if ( ! $is_dry_run ) {
+					$default_modules = Jetpack::get_default_modules();
+					Jetpack::update_active_modules( $default_modules );
+				} else {
+					WP_CLI::success( __( 'This is a `Dry Run` and module activation will be reset', 'jetpack' ) );
+				}
+
 				WP_CLI::success( __( 'Modules reset to default.', 'jetpack' ) );
 				break;
 			case 'prompt':
-				WP_CLI::error( __( 'Please specify if you would like to reset your options, or modules', 'jetpack' ) );
+				WP_CLI::error( __( 'Please specify if you would like to reset your options, modules or sync-checksum', 'jetpack' ) );
 				break;
+			case 'sync-checksum':
+				global $wpdb;
+				$option = 'jetpack_callables_sync_checksum';
+
+				if ( $is_dry_run ) {
+					WP_CLI::success( __( 'This is a `Dry Run` and no options will be deleted', 'jetpack' ) );
+				}
+
+				if ( is_multisite() && function_exists( 'get_sites' ) ) {
+					$sites  = get_sites( array( 'number' => 1000 ) );
+					$count_fixes  = 0;
+					foreach ( $sites as $site ) {
+						switch_to_blog( $site->blog_id );
+						$count = self::count_option( $option );
+						if ( $count > 1 ) {
+							if ( $is_dry_run ) {
+								WP_CLI::line( sprintf( __( 'DRY RUN: Deleted %s %s options from %s', 'jetpack' ), $count, $option, "{$site->domain}{$site->path}" ) );
+							} else {
+								delete_option( $option );
+								WP_CLI::line( sprintf( __( 'Deleted %s %s options from %s', 'jetpack' ), $count, $option, "{$site->domain}{$site->path}" ) );
+							}
+
+							$count_fixes++;
+							$sleep_duration = ( $is_dry_run ? 1 : 20 );
+							sleep( $sleep_duration ); // Allow some time for replication to catch up.
+						}
+
+						restore_current_blog();
+					}
+					if ( $count_fixes ) {
+						WP_CLI::success( sprintf( __( "Successfully reset %s on %s sites.", 'jetpack' ), $option, $count_fixes ) );
+					} else {
+						WP_CLI::success( __( "No options were deleted.", 'jetpack' ) );
+					}
+					if ( $is_dry_run ) {
+						WP_CLI::line( "\n" . __( "This was a DRY RUN!", 'jetpack' ) );
+					}
+					return;
+				}
+
+				$count = self::count_option( $option );
+				if ( $count > 1 ) {
+					if ( $is_dry_run ) {
+						WP_CLI::success( sprintf( __( 'DRY RUN: Deleted %s %s options', 'jetpack' ), $count, $option ) );
+					} else {
+						delete_option( $option );
+						WP_CLI::success( sprintf( __( 'Deleted %s %s options', 'jetpack' ), $count, $option ) );
+					}
+
+					return;
+				}
+
+				WP_CLI::success( __( "No options were deleted.", 'jetpack' ) );
+				break;
+
 		}
+		if ( $is_dry_run ) {
+			WP_CLI::line( "\n" . __( "This was a DRY RUN!", 'jetpack' ) );
+		}
+	}
+
+	static function count_option( $option ) {
```

Comment:
Should this be plural, as we might be counting multiple options?

```suggestion
	static function count_options( $option ) {
```

Reply:
@tyxla isn't the method necessarily counting a single option as per the sql query? `return (int) $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM $wpdb->options WHERE option_name = %s", $option ) );`

Reply:
@gititon please correct me if I'm wrong, but the idea with this suggestion is related to the fact  that we can accidentally create multiple `jetpack_callables_sync_checksum` options, and this method counts how many of them we have currently. So if we're counting multiple things, plural seems like an obvious choice. Let me know if that makes sense.

Reply:
@tyxla I interpret `count_option` as `count the instances of this particular option`, which is what the method does, but I am happy to see it named however the team likes.

Reply:
My vote is for `count_option`

Reply:
👍 Let's keep it as-is then 👍